File: Tyche.Android/data/bet/src/main/java/com/felipearpa/data/bet/domain/PoolGamblerBetRepository.kt
package com.felipearpa.data.bet.domain

import com.felipearpa.tyche.core.paging.CursorPage

interface PoolGamblerBetRepository {
    suspend fun bet(bet: Bet): Result<PoolGamblerBet>

    suspend fun getPendingPoolGamblerBets(
        poolId: String,
        gamblerId: String,
        next: String? = null,
        searchText: String? = null
    ): Result<CursorPage<PoolGamblerBet>>

    suspend fun getFinishedPoolGamblerBets(
        poolId: String,
        gamblerId: String,
        next: String? = null,
        searchText: String? = null
    ): Result<CursorPage<PoolGamblerBet>>
}

File: Tyche.iOS/DataBet/Sources/DataBet/Domain/PoolGamblerBetRepository.swift
import Core

public protocol PoolGamblerBetRepository {
    func getPendingPoolGamblerBets(
        poolId: String,
        gamblerId: String,
        next: String?,
        searchText: String?
    ) async -> Result<CursorPage<PoolGamblerBet>, Error>

    func getFinishedPoolGamblerBets(
        poolId: String,
        gamblerId: String,
        next: String?,
        searchText: String?
    ) async -> Result<CursorPage<PoolGamblerBet>, Error>

    func bet(bet: Bet) async -> Result<PoolGamblerBet, Error>
}
